doc/source/serve/production-guide/config.md

```diff
@@ -0,0 +1,200 @@
+  user_config:
+    price: 4
+  autoscaling_config: null
+  graceful_shutdown_wait_loop_s: 2.0
+  graceful_shutdown_timeout_s: 20.0
+  health_check_period_s: 10.0
+  health_check_timeout_s: 30.0
+  ray_actor_options: null
+
+- name: FruitMarket
+  num_replicas: 2
+  route_prefix: null
+  max_concurrent_queries: 100
+  user_config: null
+  autoscaling_config: null
+  graceful_shutdown_wait_loop_s: 2.0
+  graceful_shutdown_timeout_s: 20.0
+  health_check_period_s: 10.0
+  health_check_timeout_s: 30.0
+  ray_actor_options: null
+
+- name: DAGDriver
+  num_replicas: 1
+  route_prefix: /
+  max_concurrent_queries: 100
+  user_config: null
+  autoscaling_config: null
+  graceful_shutdown_wait_loop_s: 2.0
+  graceful_shutdown_timeout_s: 20.0
+  health_check_period_s: 10.0
+  health_check_timeout_s: 30.0
+  ray_actor_options: null
+
+```
+
+Note that the `runtime_env` field will always be empty when using `serve build` and must be set manually.
+
+## Overriding deployment settings
+
+Settings from `@serve.deployment` can be overriden with this Serve config file. The order of priority is (from highest to lowest):
+
+1. Config File
+2. Deployment graph code (either through the `@serve.deployment` decorator or a `.set_options()` call)
+3. Serve defaults
+
+For example, if a deployment's `num_replicas` is specified in the config file and their graph code, Serve will use the config file's value. If it's only specified in the code, Serve will use the code value. If the user doesn't specify it anywhere, Serve will use a default (which is `num_replicas=1`).
+
+Keep in mind that this override order is at the settings-level. For example, if a user has a deployment `ExampleDeployment` with the following decorator:
+
+```python
+@serve.deployment(
+    num_replicas=2,
+    max_concurrent_queries=15,
+)
+class ExampleDeployment:
+    ...
+```
+
+and the following config file:
+
+```yaml
+...
+
+deployments:
+
+    - name: ExampleDeployment
+      num_replicas: 5
+
+...
+```
+
+Serve will set `num_replicas=5`, using the config file value, and `max_concurrent_queries=15`, using the code value (since `max_concurrent_queries` wasn't specified in the config file). All other deployment settings use Serve defaults since the user didn't specify them in the code or the config.
+
+:::{tip}
+Remember that `ray_actor_options` is an independent setting. The entire `ray_actor_options` dictionary in the config file overrides the entire `ray_actor_options` dictionary from the graph code. If there are individual options within `ray_actor_options` (e.g. `runtime_env`, `num_gpus`, `memory`) that are set in the code but not in the config, Serve still won't use the code settings if the config has a `ray_actor_options` dictionary. It will treat these missing options as though the user never set them (and use defaults instead) since the entire `ray_actor_options` dictionary in the config overrides the one in the code. This dictionary overriding behavior also applies to `user_config`.
```

Comment:
```suggestion
Remember that `ray_actor_options` counts as a single setting. The entire `ray_actor_options` dictionary in the config file overrides the entire `ray_actor_options` dictionary from the graph code. If there are individual options within `ray_actor_options` (e.g. `runtime_env`, `num_gpus`, `memory`) that are set in the code but not in the config, Serve still won't use the code settings if the config has a `ray_actor_options` dictionary. It will treat these missing options as though the user never set them and will use defaults instead. This dictionary overriding behavior also applies to `user_config`.
```